core/src/apps/management/recovery_device/layout.py

```diff
@@ -45,13 +42,11 @@ async def request_word_count(ctx: wire.GenericContext, dry_run: bool) -> int:
         text = Text("Recovery mode", ui.ICON_RECOVERY)
     text.normal("Number of words?")
 
-    if __debug__:
-        count = await ctx.wait(WordSelector(text), input_signal())
-        count = int(count)  # if input_signal was triggered, count is a string
-    else:
-        count = await ctx.wait(WordSelector(text))
-
-    return count  # type: ignore
+    count = await ctx.wait(WordSelector(text))
+    # WordSelector can return int, or string if the value came from debuglink
+    # ctx.wait has a return type Any
+    # Hence, it is easier to convert the returned value to int explicitly
+    return int(count)
```

Comment:
A small off-topic: we could add another field to DebugLinkDecision `input_type` and casting the `input` to the appropriate type before sending to the input_chan. What do you think? (Let's not do it here, I would open an issue or do it after merge.)

Although the comment clarifies it perfectly, it is still a bit weird.

Reply:
ISTM that would cause more complication than clarity.

The above code could have been otherwise solved by casting the DebugLinkDecision result to int inside the WordSelector -- that's arguably a more correct solution, as it abstracts away the existence of debuglink on the correct level.
The reason I did it this way is because of `ctx.wait` returning `Any` anyway -- so casting the result beforehand would have no practical advantage and i'd still need to do `int(count)` here.

Adding input_type to DebugLinkDecision would move the type awareness away from the right place (which is WordSelector) and away from the core codebase entirely.